cloud/deployment/deployment.go

```diff
@@ -374,16 +375,19 @@ func Create(name, workspaceID, description, clusterID, runtimeVersion, dagDeploy
 				IsDevelopmentMode:    &developmentModeValue,
 				WorkspaceId:          workspaceID,
 				Type:                 astroplatformcore.CreateStandardDeploymentRequestTypeSTANDARD,
-				DefaultTaskPodCpu:    &defaultTaskPodCpu,
-				DefaultTaskPodMemory: &defaultTaskPodMemory,
-				ResourceQuotaCpu:     &resourceQuotaCpu,
-				ResourceQuotaMemory:  &resourceQuotaMemory,
+				DefaultTaskPodCpu:    defaultTaskPodCPUPtr,
+				DefaultTaskPodMemory: defaultTaskPodMemoryPtr,
+				ResourceQuotaCpu:     resourceQuotaCPUPtr,
+				ResourceQuotaMemory:  resourceQuotaMemoryPtr,
 				WorkloadIdentity:     deplWorkloadIdentity,
+				RemoteExecution:      remoteExecution,
 			}
-			if strings.EqualFold(executor, CeleryExecutor) || strings.EqualFold(executor, CELERY) || strings.EqualFold(executor, AstroExecutor) || strings.EqualFold(executor, ASTRO) {
-				standardDeploymentRequest.WorkerQueues = &defautWorkerQueue
+			if standardDeploymentRequest.RemoteExecution == nil || !standardDeploymentRequest.RemoteExecution.Enabled {
+				if strings.EqualFold(executor, CeleryExecutor) || strings.EqualFold(executor, CELERY) || strings.EqualFold(executor, AstroExecutor) || strings.EqualFold(executor, ASTRO) {
+					standardDeploymentRequest.WorkerQueues = &defautWorkerQueue
+				}
 			}
-			switch schedulerSize {
+			switch schedulerSize { //nolint:dupl
```

Comment:
Should this be a utility method instead of using nolint:dupl?

Reply:
ok, I think there is scope to improve still, generics could be one way to look at it. But not a blocker for this PR, as I already have a story incoming from the other nova to make these insane if/else blocks a bit simpler.